### PR TITLE
Handle structure dtypes with titles

### PR DESCRIPTION
**Pull request by sklam**
Fix issue #2020

### CODE REVIEW

File: numba/tests/test_record_dtype.py [hunk 1]
```diff
@@ -663,6 +663,37 @@ def test_record_two_arrays(self):
         cfunc = self.get_cfunc(pyfunc, (nbrecord,))
         self.assertEqual(cfunc(rec), pyfunc(rec))
 
+    def test_structure_dtype_with_titles(self):
+        # the following is the definition of int4 vector type from pyopencl
+        vecint4 = np.dtype([(('x', 's0'), '<i4'), (('y', 's1'), '<i4'),
```
Comment (pitrou) [inline numba/tests/test_record_dtype.py]: Just curious: `<i4` won't work a big-endian machine since we only support native-endian arrays, right?

Comment (sklam) [inline numba/tests/test_record_dtype.py]: right, and I don't think we are checking the endian at all.

Comment (pitrou) [inline numba/tests/test_record_dtype.py]: So the test could simply use `'i4'` instead of `'<i4'`?

Comment (sklam) [inline numba/tests/test_record_dtype.py]: Sure

**Pull request merged**